ipykernel/zmqshell.py

```diff
@@ -91,6 +125,48 @@ def clear_output(self, wait=False):
             parent=self.parent_header, ident=self.topic,
         )
 
+    def register_hook(self, hook):
+        """
+        Registers a hook with the thread-local storage.
+
+        Parameters
+        ----------
+        hook : IPython.core.displayhook.DisplayHook
+               An instance of DisplayHook.
+
+        Returns
+        -------
+        None
+
+        The DisplayHook objects must return a message from
```

Comment:
I don't think these are DisplayHook objects since DisplayHook is called on objects, not messages. I think it's any callable with:

```python
def hook(message) => message or None
```